### PR TITLE
fix: tmclient update header validation for v0.44.x

### PR DESCRIPTION
**Pull request by rnbguy**
Closes: #911 but on v0.44.x release

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests.
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: .changelog/unreleased/bug/911-fix-header-validation-during-tmclient-update.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- Remove an incorrect validation during tendermint client update
```
Comment (Farhad-Shabani) [inline .changelog/unreleased/bug/911-fix-header-validation-during-tmclient-update.md:1]: Should this be placed under the `.changelog/v0.44.2/bug-fixes` ?

Comment (rnbguy) [reply .changelog/unreleased/bug/911-fix-header-validation-during-tmclient-update.md:1]: I wanted to make the changes on `release/v0.44.x` (based on `tags/v0.44.1`). You also have to create the other PR on `release/v0.44.x`. Then, perform a patch release on `release/v0.44.x` - which bumps the version from `0.44.1` to `0.44.2` and moves the changelog entries under `.changelog/v0.44.2`.

Does it make sense?

Comment (Farhad-Shabani) [reply .changelog/unreleased/bug/911-fix-header-validation-during-tmclient-update.md:1]: Oh, I see. Gotcha 👍🏻

**Review by Farhad-Shabani — APPROVED** [review]
Thanks for taking care of this @rnbguy!

**Pull request merged**